ref(helm): expose Get for repository file

hypnoglow opened the pull request:
Get method is a useful shortcut that can be used by other apps that use helm repo package.

bacongobbler commented:
hey @hypnoglow, can you explain this PR in more detail? I'm a little confused on what this addresses/fixes. Thanks!

hypnoglow commented:
Hi @bacongobbler ! 

> can you explain this PR in more detail? I'm a little confused on what this addresses/fixes.

This is just a convenience that may helpful for third-party plugins using helm. For example, [here](https://github.com/hypnoglow/helm-s3/blob/master/internal/helmutil/repo_entry.go#L34) in my helm-s3 plugin I have to manually iterate over repositories to get an entry. But with this shortcut I could just use `repoFile.Get(name)`.

> It appears that `.Get` is already a function in this package. 

There is no `Get` method for `RepoFile`, but for `Index`.

bacongobbler commented:
Okay thanks for the clarification!

Can you explain in more detail why we need to update `.Remove` as well? Everything else looks fine, but the changes to `.Remove` should probably be split into a separate PR so we can review it separately. Would you mind doing that? Thanks!

hypnoglow committed:
ref(helm): expose Get for repository file

Get method is a useful shortcut that can be used
by other apps that use helm repo package.

hypnoglow commented:
@bacongobbler Now this PR contains only changes related to `Get`

hypnoglow commented:
/remove-lifecycle rotten

bacongobbler reviewed (approved):
Hi @hypnoglow, sorry for leaving this one dangling for so long. After reviewing this PR, the functionality looks great, and I'm happy to see that you added unit tests to this PR. Thanks so much!

bacongobbler commented:
I'm going ahead and merging this PR as it was contributed before Helm moved to a DCO, and the contributor signed the CLA at that time. Thanks!
